Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Management/Driver.hs

```diff
@@ -1249,3 +1252,29 @@ checkDriverOperatorAssociation :: (EsqDBFlow m r, MonadFlow m, CacheFlow m r) =>
 checkDriverOperatorAssociation driverId operatorId = do
   mbAssoc <- QDriverOperator.findByDriverIdAndOperatorId driverId operatorId True
   return $ isJust mbAssoc
+
+getDriverEarnings :: ShortId DM.Merchant -> Context.City -> Day -> Day -> Common.EarningType -> Id Common.Driver -> Text -> Flow Common.EarningPeriodStatsRes
+getDriverEarnings merchantShortId opCity from to earningType dId requestorId = do
+  merchant <- findMerchantByShortId merchantShortId
+  merchantOpCityId <- CQMOC.getMerchantOpCityId Nothing merchant (Just opCity)
+  let driverId = cast @Common.Driver @DP.Person dId
+  entities <- QPerson.findAllByPersonIdsAndMerchantOpsCityId [Id requestorId, driverId] merchantOpCityId
+  driver <- find (\e -> e.id == driverId) entities & fromMaybeM (PersonDoesNotExist driverId.getId)
+  requestor <- find (\e -> e.id == Id requestorId) entities & fromMaybeM (PersonDoesNotExist requestorId)
+  isValid <- validateDriverAccessAndAssociation [requestor, driver]
```

Comment:
`validateDriverAccessAndAssociation` can be removed

`isValid <- isAssociationWithDriver requestor  driver`